common/Resources/ti.internal/extensions/node/internal/errors.js

```diff
@@ -0,0 +1,205 @@
+	if (assert === undefined) {
+		assert = require('./internal/assert');
+	}
+	*/
+
+	if (typeof msg === 'function') {
+		assert(
+			msg.length <= args.length, // Default options do not count.
+			`Code: ${key}; The provided arguments length (${args.length}) does not ` +
+				`match the required ones (${msg.length}).`
+		);
+		return msg.apply(self, args);
+	}
+
+	const expectedLength = (msg.match(/%[dfijoOs]/g) || []).length;
+	assert(
+		expectedLength === args.length,
+		`Code: ${key}; The provided arguments length (${args.length}) does not ` +
+			`match the required ones (${expectedLength}).`
+	);
+	if (args.length === 0) {
+		return msg;
+	}
+
+	args.unshift(msg);
+	return format.apply(null, args);
+	// @fixme rollup cannot handle lazy loaded modules, maybe move to webpack?
+	// return lazyInternalUtilInspect().format.apply(null, args);
+}
+
+function addCodeToName(err, name, code) {
+	// Add the error code to the name to include it in the stack trace.
+	err.name = `${name} [${code}]`;
+	// Access the stack to generate the error message including the error code
+	// from the name.
+	// @fixme: This only works on V8/Android, iOS/JSC has a different Error structure.
+	// should we try to make errors behave the same across platforms?
+	err.stack;
+	// Reset the name to the actual name.
+	if (name === 'SystemError') {
+		Object.defineProperty(err, 'name', {
+			value: name,
+			enumerable: false,
+			writable: true,
+			configurable: true
+		});
+	} else {
+		delete err.name;
+	}
+}
+
+E('ERR_INTERNAL_ASSERTION', (message) => {
+	const suffix = 'This is caused by either a bug in Titanium ' +
+		'or incorrect usage of Titanium internals.\n' +
+		'Please open an issue with this stack trace at ' +
+		'https://jira.appcelerator.org\n';
+	return message === undefined ? suffix : `${message}\n${suffix}`;
+}, Error);
+E('ERR_INVALID_ARG_TYPE', (name, expected, actual) => {
+	assert(typeof name === 'string', '\'name\' must be a string');
+
+	// determiner: 'must be' or 'must not be'
+	let determiner;
+	if (typeof expected === 'string' && expected.startsWith('not ')) {
+		determiner = 'must not be';
+		expected = expected.replace(/^not /, '');
+	} else {
+		determiner = 'must be';
+	}
+
+	let msg;
+	if (name.endsWith(' argument')) {
+		// For cases like 'first argument'
+		msg = `The ${name} ${determiner} ${oneOf(expected, 'type')}`;
+	} else {
+		const type = name.includes('.') ? 'property' : 'argument';
+		msg = `The "${name}" ${type} ${determiner} ${oneOf(expected, 'type')}`;
+	}
+
+	// TODO(BridgeAR): Improve the output by showing `null` and similar.
+	msg += `. Received type ${typeof actual}`;
+	return msg;
+}, TypeError);
+
+let maxStack_ErrorName;
+let maxStack_ErrorMessage;
+/**
+ * Returns true if `err.name` and `err.message` are equal to engine-specific
+ * values indicating max call stack size has been exceeded.
+ * "Maximum call stack size exceeded" in V8.
+ *
+ * @param {Error} err
```

Comment:
<!--
  0 failure: 
  1 warning:  common/Resources/...
  
  
  DangerID: danger-id-Danger;
  File: common/Resources/ti.internal/extensions/node/internal/errors.js;
  Line: 170;
-->

- :warning: common/Resources/ti.internal/extensions/node/internal/errors.js line 170 – Missing JSDoc parameter description for 'err'. (valid-jsdoc)